### PR TITLE
Refactor deploy script for extensions

### PR DESCRIPTION
**Pull request by samansmink**
This PR does a small refactor with as a goal:

- deduplicate the `.github/workflows/_extension_deploy.yml` that we currently have duplicated across all out-of-tree extensions
- deduplicate the `scripts/extension-upload-single.sh` that we currently have duplicated across all out-of-tree extensions
- unify the script we use to deploy out of tree extensions to nightly and the production ones in the this repo
- remove some unused stuff i came across

Maybe @carlopi can chime in here on the WASM part here?

A note on the deduplication of the two files: we can only use those from within the duckdb organisation because we need to pass secrets into them. 3rd party will still need to copy the scripts

### CODE REVIEW

File: scripts/extension-upload-single.sh [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+truncate -s 256 $ext.sign
+
+# append signature to extension binary
+cat $ext.sign >> $ext.append
+
+# compress extension binary
+if [[ $4 == wasm_* ]]; then
+  gzip < $ext.append > "$ext.compressed"
+else
+  brotli < $ext.append > "$ext.compressed"
```
Comment (carlopi) [inline scripts/extension-upload-single.sh:70]:
```suggestion
  brotli < $ext.append > "$ext.compressed"
else
  gzip < $ext.append > "$ext.compressed"
```

**Review by carlopi — CHANGES_REQUESTED** [review]
There is a change needed (error initially introduced by me that keeps popping up), but the rest looks solid.

Thanks

File: scripts/extension-upload-single.sh (hunk 1 above)
Comment (samansmink) [reply scripts/extension-upload-single.sh]: oops, good catch!

**Comment by samansmink** [comment]
@Mytherin I think this is good to go, CI doesn't really do much for this PR

**Comment by djouallah** [comment]
@samansmink can you please update the iceberg extension :)

**Pull request merged**